File: src/org/nextras/orm/intellij/annotator/highlighter/ModifierHighlighterSettings.java
package org.nextras.orm.intellij.annotator.highlighter;

import com.intellij.openapi.editor.colors.TextAttributesKey;
import com.intellij.openapi.fileTypes.SyntaxHighlighter;
import com.intellij.openapi.options.colors.AttributesDescriptor;
import com.intellij.openapi.options.colors.ColorDescriptor;
import com.intellij.openapi.options.colors.ColorSettingsPage;
import com.jetbrains.php.lang.highlighter.PhpFileSyntaxHighlighter;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;
import org.nextras.orm.intellij.Icons;

import javax.swing.*;
import java.util.Map;


public class ModifierHighlighterSettings implements ColorSettingsPage
{
	private static final AttributesDescriptor[] DESCRIPTORS = new AttributesDescriptor[]{
			new AttributesDescriptor("Modifier", ModifierHighlighter.MODIFIER),
			new AttributesDescriptor("Brackets", ModifierHighlighter.BRACKETS),
	};


	@Nullable
	@Override
	public Icon getIcon()
	{
		return Icons.FILE;
	}


	@NotNull
	@Override
	public SyntaxHighlighter getHighlighter()
	{
		return new PhpFileSyntaxHighlighter();
	}


	@NotNull
	@Override
	public String getDemoText()
	{
		return "<?php\n// Sadly, there is no way how to preview your color settings in here.";
	}


	@Nullable
	@Override
	public Map<String, TextAttributesKey> getAdditionalHighlightingTagToDescriptorMap()
	{
		return null;
	}


	@NotNull
	@Override
	public AttributesDescriptor[] getAttributeDescriptors()
	{
		return DESCRIPTORS;
	}


	@NotNull
	@Override
	public ColorDescriptor[] getColorDescriptors()
	{
		return ColorDescriptor.EMPTY_ARRAY;
	}


	@NotNull
	@Override
	public String getDisplayName()
	{
		return "Nextras Orm";
	}
}

File: src/main/kotlin/org/nextras/orm/intellij/annotator/highlighter/ModifierHighlighterSettings.kt
package org.nextras.orm.intellij.annotator.highlighter

import com.intellij.openapi.editor.colors.TextAttributesKey
import com.intellij.openapi.fileTypes.SyntaxHighlighter
import com.intellij.openapi.options.colors.AttributesDescriptor
import com.intellij.openapi.options.colors.ColorDescriptor
import com.intellij.openapi.options.colors.ColorSettingsPage
import com.jetbrains.php.lang.highlighter.PhpFileSyntaxHighlighter
import javax.swing.Icon

class ModifierHighlighterSettings : ColorSettingsPage {
	override fun getIcon(): Icon? {
		return null
	}

	override fun getHighlighter(): SyntaxHighlighter {
		return PhpFileSyntaxHighlighter()
	}

	override fun getDemoText(): String {
		return "<?php\n// IntelliJ IDEs do not support previewing custom highlighter here."
	}

	override fun getAdditionalHighlightingTagToDescriptorMap(): Map<String, TextAttributesKey>? {
		return null
	}

	override fun getAttributeDescriptors(): Array<AttributesDescriptor> {
		return DESCRIPTORS
	}

	override fun getColorDescriptors(): Array<ColorDescriptor> {
		return ColorDescriptor.EMPTY_ARRAY
	}

	override fun getDisplayName(): String {
		return "Nextras Orm"
	}

	companion object {
		private val DESCRIPTORS = arrayOf(
			AttributesDescriptor("Modifier braces", ModifierHighlighter.BRACES),
			AttributesDescriptor("Modifier name", ModifierHighlighter.MODIFIER),
			AttributesDescriptor("Modifier identifier", ModifierHighlighter.IDENTIFIER)
		)
	}
}
